src/sqlfluff/dialects/dialect_sparksql.py

```diff
@@ -3113,3 +3113,20 @@ class SelectClauseSegment(BaseSegment):
     )
 
     parse_grammar: Matchable = Ref("SelectClauseSegmentGrammar")
+
+
+class CTEDefinitionSegment(ansi.CTEDefinitionSegment):
+    """A CTE Definition from a WITH statement.
+
+    `tab (col1,col2) AS (SELECT a,b FROM x)`
+    """
+
+    match_grammar: Matchable = Sequence(
+        Ref("SingleIdentifierGrammar"),
+        Ref("CTEColumnList", optional=True),
+        Sequence("AS", optional=True),
```

Comment:
I think it's ok to just make this change in `dialect_ansi` to make "AS" optional instead of overriding `CTEDefinitionSegment` here.

Reply:
Hey @greg-finley, I believe not all dialects has the  `AS` as optional keyword for CTE queries In my use case I tried with mysql, hive, redshift, clickhouse & oracle and CTE without `AS` keyword is not supported!

Reply:
It’s ok to make the ANSI file a little looser than the dialects themselves as long as SQL still parses. It should in this case, and I feel it’s better than copying the segment into this dialect.